pkg/grpc/codec_test.go

```diff
@@ -0,0 +1,99 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+	"google.golang.org/protobuf/types/known/durationpb"
+)
+
+func TestCodec(t *testing.T) {
+	svc := getClientForServer(t)
+
+	// create a small message that will be below (<=)
+	// buffer pooling threshold
+	request := v1.SuppressCVERequest{
```

Comment:
Prefer relying on a test-only service and proto structure.

Reply:
we do not have any service like that. I think this is quite small and we are using mock implementation that returns error always so there is 0 chance of changes (unless new method is added) as we expect backward compatibility.